apps/sv/frontend/src/components/ValidatorOnboardingSecrets.tsx

```diff
@@ -57,58 +76,152 @@ const ValidatorOnboardingSecrets: React.FC = () => {
         </Typography>
       )}
 
+      <Stack direction="column" mb={4} spacing={1}>
+        <Typography variant="h6">Party Hint</Typography>
+        <TextField
+          error={partyHint === '' || !VALID_PARTY_ID_REGEX.test(partyHint)}
+          autoComplete="off"
+          id="create-party-hint"
+          placeholder="<organization>-<function>-<enumerator>"
+          inputProps={{ 'data-testid': 'create-party-hint' }}
+          onChange={e => setPartyHint(e.target.value)}
+          value={partyHint}
+        />
+      </Stack>
+
       <DisableConditionally
-        conditions={[{ disabled: prepareOnboardingMutation.isPending, reason: 'Loading...' }]}
+        conditions={[
+          { disabled: prepareOnboardingMutation.isPending, reason: 'Loading...' },
+          { disabled: partyHint === '', reason: 'No Party Hint' },
+          {
+            disabled: !VALID_PARTY_ID_REGEX.test(partyHint),
+            reason: 'Party Hint is invalid',
+            severity: 'warning',
+          },
+        ]}
       >
         <Button
           id="create-validator-onboarding-secret"
+          data-testid="create-validator-onboarding-secret"
           variant="pill"
           fullWidth
-          disabled={prepareOnboardingMutation.isPending}
           size="large"
-          onClick={() => prepareOnboardingMutation.mutate()}
+          onClick={() => {
+            prepareOnboardingMutation.mutate(partyHint);
+            setPartyHint('');
```

Comment:
Probably not important but doesn't this clear the field if the request fails? I would have expected to mutateAsync and clear on success, show error on failure.

Reply:
I think on failure, you  actually end up on the error boundary page. Not sure how important that is though